File: 프로그래머스/LV2/코딩테스트 연습/탐욕법/조이스틱/조이스틱.py
def solution(name):
    answer = 0
    name_len = len(name)
    min_move = name_len - 1
    
    for i in range(name_len):
        answer += min(ord(name[i]) - ord('A'), ord('Z') + 1 - ord(name[i]))
        idx = i + 1

        while idx < name_len and name[idx] == 'A':
            idx += 1

        min_move = min(min_move, i * 2 + name_len - idx)
        min_move = min(min_move, (name_len - idx) * 2 + i)

    return answer + min_move

File: 프로그래머스/LV2/코딩테스트 연습/탐욕법/조이스틱/조이스틱.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string name) {
    int answer = 0;
    int name_len = name.length();
    int min_move = name_len - 1;
    int idx = 0;

    for(int i = 0; i < name_len; i++) {
        answer += min(name[i] - 'A', 'Z' + 1 - name[i]);
        idx = i + 1;

        while (idx < name_len && name[idx] == 'A')
            idx++;

        min_move = min(min_move, i * 2 + name_len - idx);
        min_move = min(min_move, (name_len - idx) * 2 + i);
    }
    return answer + min_move;
}